### DISCUSSION TITLE
How can I view huggingface dataset images?

### DISCUSSION THREAD

**Discussion by asmith26** [Eventual-Inc/Daft#2747] [Q&A] [answered]
Hi there, I'm trying to view a huggingface dataset including images:

```python
import daft
df = daft.read_parquet("hf://datasets/HuggingFaceM4/DocumentVQA")
df.show(5)
```

![image](https://github.com/user-attachments/assets/74b4b7b5-7c09-4e65-9a81-dee00fc57cb1)

Following https://www.getdaft.io/projects/docs/en/stable/10-min.html#working-with-multimodal-data I try:

```python
df = df.with_column("image_show", daft.col("image").image.decode())
```

Unfortunately this yields:

```python
DaftCoreException: DaftError::External Unable to create logical plan node.
Due to: DaftError::TypeError ImageDecode can only decode BinaryArrays, got image#Struct[bytes: Binary, path: Utf8]
```

It looks like the huggingface dataset image column has additional information that needs to handled first, but I can't seem to work out how to fix this?

Many thanks for any help, and this amazing lib! :)

**Comment by jaychia** [comment, marked as answer]
Oh interesting. Looks like this particular dataset stores images as some kind of struct column, with both the URL and bytes for some reason 🤷 

You can try:

```
df = df.with_column("decoded_image", daft.col("image.bytes").image.decode())
```

`.image.decode()` works on a binary column, which in this case is a nested column under the `image` struct column.

**Reply by asmith26** [reply]
Many thanks for the very quick response and help @jaychia, can confirm it works!! :)

![image](https://github.com/user-attachments/assets/a28aaaf4-7880-4a2f-84b8-9fb6e25708a5)
